src/Common/FileSegment.cpp

```diff
@@ -641,6 +665,12 @@ FileSegmentsHolder::~FileSegmentsHolder()
         if (!cache)
             cache = file_segment->cache;
 
+        if (file_segment->detached)
+        {
+            file_segment_it = file_segments.erase(current_file_segment_it);
```

Comment:
Do we need `std::lock_guard cache_lock(cache->mutex)` here? If no -- need a comment.

Reply:
Not needed, as it is not owned by cache.

Reply:
Not needed, as it is not owned by cache.